lib/open_api_spex/cast/any_of.ex

```diff
@@ -14,37 +15,51 @@ defmodule OpenApiSpex.Cast.AnyOf do
          failed_schemas,
          acc
        ) do
-    relaxed_schema = %{schema | additionalProperties: true, "x-struct": nil}
+    relaxed_schema = %{schema | "x-struct": nil}
     new_ctx = put_in(ctx.schema.anyOf, remaining)
 
-    case Cast.cast(%{ctx | schema: relaxed_schema}) do
+    case Cast.cast(%{ctx | errors: [], schema: relaxed_schema}) do
       {:ok, value} when is_map(value) ->
-        acc =
-          value
-          |> Enum.reject(fn {k, _} -> is_binary(k) end)
-          |> Enum.concat(acc || %{})
-          |> Map.new()
-
-        cast_any_of(new_ctx, failed_schemas, acc)
+        cast_any_of(new_ctx, failed_schemas, Map.merge(acc || %{}, value))
```

Comment:
```suggestion
        cast_any_of(new_ctx, failed_schemas, merge_maps(acc || %{}, value))
```

where
```elixir
defp merge_maps(map1, map2) do
  l1 = Enum.map(map1, fn {key, val} -> {to_string(key), key, val} end)
  l2 = Enum.map(map2, fn {key, val} -> {to_string(key), key, val} end)

  l1
  |> Kernel.++(l2)
  |> Enum.group_by(fn {ks, _, _} -> ks end, fn {_, k, v} -> {k, v} end)
  |> Map.new(fn
    {_ks, [{k, v}]} ->
      {k, v}

    {_ks, [{k1, v1}, {k2, v2}]} ->
      cond do
        is_atom(k2) -> {k2, v2}
        is_atom(k1) -> {k1, v1}
        true -> {k2, v2}
      end
  end)
end
```
This function merges 2 maps considering equal keys that are an atom or a string representation of that atom.

This is needed for avoiding situations like the following
```elixir
defmodule A do
  alias OpenApiSpex.{Cast, Schema}
  alias OpenApiSpex.Cast.AnyOf

  defp cast(ctx), do: AnyOf.cast(struct(Cast, ctx))

  def run(value) do
    dog_schema = %Schema{
      title: "Dog",
      type: :object,
      properties: %{
        breed: %Schema{type: :string},
        age: %Schema{type: :integer}
      }
    }

    cat_schema = %Schema{type: :object}

    schema = %Schema{anyOf: [dog_schema, cat_schema], title: "MyCoolSchema"}
    cast(value: value, schema: schema)
  end
end
```
that with the current implementation returns
```
iex(1)> A.run(%{"breed" => "aaa", "age" => 1, "foo" =>"bar"})
{:ok,
 %{:age => 1, :breed => "aaa", "age" => 1, "breed" => "aaa", "foo" => "bar"}}
```
instead of
```
iex(1)> A.run(%{"breed" => "aaa", "age" => 1, "foo" =>"bar"})
{:ok, %{:age => 1, :breed => "aaa", "foo" => "bar"}}
```